tests/test_impute.py

```diff
@@ -0,0 +1,97 @@
+from nursepy.impute import impute 
+import numpy as np
+import pandas as pd
+
+
+def random_data():
+    """
+    Generates random data for testing impute function
+
+    Return
+    ------
+    A six element tuple of pandas dataframes
+    """
+    Xt = {'one': np.random.randn(10),
+        'two': np.random.randn(10),
+        'three': np.random.randn(10),
+        'four': np.random.randn(10),
+        'five': np.random.randn(10)}
+    yt_n = pd.DataFrame({'target': np.random.randn(10)})
+    yt_c = pd.DataFrame({'target': [1,0,1,1,0,0,0,1,0,1]})
+
+    Xt['two'][3:5] = None
+    Xt['three'][7] = None
+    Xt['four'][1] = None
+    Xt['four'][3] = None
+    Xt['five'][2] = None
+    Xt = pd.DataFrame(Xt)
+
+    Xv = {'one': np.random.randn(10),
+        'two': np.random.randn(10),
+        'three': np.random.randn(10),
+        'four': np.random.randn(10),
+        'five': np.random.randn(10)}
+    yv_n = pd.DataFrame({'target': np.random.randn(10)})
+    yv_c = pd.DataFrame({'target': [1,0,0,0,1,1,1,0,0,1]})
+
+    Xv['one'][2:4] = None
+    Xv['two'][2] = None
+    Xv['four'][1] = None
+    Xv['four'][4] = None
+    Xv['five'][8] = None
+    Xv = pd.DataFrame(Xv)
+
+    return (Xt, yt_n, yt_c, Xv, yv_n, yv_c)
+
+def test_blocks():
+    """
+    Tests that all blocks in impute function are satisfied.
+    A sequence of try-except clauses capturing UserErrors in impute
+    Raises assertion errors if tests fail
+    """
+    Xt, yt_n, yt_c, Xv, yv_n, yv_c = random_data()
+    
+    Xt1 = Xt.copy()
+    Xt1['five'] = 'string'
+
+    try:
+        impute(dict(Xt1), yt_c, Xv, yv_c, model_type = 'classification')
+    except Exception as exc:
+        assert(exc.args[0] == 'pandas dataframe objects are required')
+
+    try:
```

Comment:
I think there is potentially issues with testing for errors in this way. If an error is not thrown, the test will pass

Reply:
thanks Elliott - that is the point of the assert though - it's looking for the error to be thrown in impute, if it wasn't thrown, then this test should fail and will throw an assertion error

Reply:
Because the assert statement will never run. Rather this test is saying, if there is an error it must be this type of error, if there isnt an error thats also fine.

Reply:
issue resolved, nice work